source/includes/fundamentals/code-snippets/SearchIndexMethods.java

```diff
@@ -0,0 +1,74 @@
+package org.example;
+
+import com.mongodb.client.*;
+import com.mongodb.client.model.SearchIndexModel;
+import org.bson.BsonBoolean;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.Document;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class SearchIndexMethods {
+
+    private static String URI = "mongodb://localhost:59651/?directConnection=true&serverSelectionTimeoutMS=2000";
+    private static String DB_NAME = "test_db";
+    private static String COLL_NAME = "test_coll";
+    private static void setup(MongoClient mongoClient) {
+        MongoDatabase database = mongoClient.getDatabase(DB_NAME);
+        MongoCollection<Document> collection = database.getCollection(COLL_NAME);
+        collection.dropIndexes();
+        database.createCollection(COLL_NAME);
+    }
+    public static void main( String[] args ) {
+
+        try (MongoClient mongoClient = MongoClients.create(URI)) {
+            setup(mongoClient);
+            MongoDatabase database = mongoClient.getDatabase(DB_NAME);
+            MongoCollection<Document> collection = database.getCollection(COLL_NAME);
+
+            // start create-search-index
+            BsonDocument index = new BsonDocument("mappings",
+                    new BsonDocument("dynamic", BsonBoolean.TRUE));
+            collection.createSearchIndex("myIndex", index);
+            // end create-search-index
+
+            // start create-search-indexes
+            SearchIndexModel indexOne = new SearchIndexModel("myIndex1",
```

Comment:
In all instances/examples, you might as well consider using `Document `instead of `BsonDocument` for simplicity. The second example avoids the explicit use of `BsonString` and `BsonBoolean` classes, which makes the code a bit simpler to comprehend.
```
        SearchIndexModel indexOne = new SearchIndexModel("myIndex1",
                new Document("analyzer", "lucene.standard")
                        .append("mappings", new Document("dynamic", true)))"
```